core/src/main/scala/org/apache/spark/api/r/RRDD.scala

```diff
@@ -127,6 +127,13 @@ private[r] object RRDD {
       sparkConf.setExecutorEnv(name.toString, value.toString)
     }
 
+    if (sparkEnvirMap.containsKey("spark.r.sql.default.derby.dir") &&
```

Comment:
Introduce "spark.sql.derby.system.home" in SQLConf as an internal config? Set the default to `System.getProperty("derby.system.home")`? Then, in the R, we can set "derby.system.home" to `tempdir()`?

Does it sound ok? @cloud-fan @rxin